src/components/container/NotificationList/NotificationList.tsx

```diff
@@ -1,91 +1,102 @@
-import React, { useState, useEffect, useRef } from 'react'
-import MyDataHelps,  { Guid, Notification, NotificationQueryParameters, NotificationType } from "@careevolution/mydatahelps-js"
-import { LoadingIndicator, SingleNotification, Card, TextBlock } from '../../presentational'
-import { previewNotifications } from './NotificationList.previewdata'
+import React, { useState } from 'react';
+import MyDataHelps, { Guid, Notification, NotificationQueryParameters, NotificationType } from '@careevolution/mydatahelps-js';
+import { Card, LoadingIndicator, SingleNotification, TextBlock } from '../../presentational';
+import { previewNotifications } from './NotificationList.previewdata';
 import language from '../../../helpers/language';
-import OnVisibleTrigger from '../../presentational/OnVisibleTrigger/OnVisibleTrigger'
+import OnVisibleTrigger from '../../presentational/OnVisibleTrigger/OnVisibleTrigger';
+import { useInitializeView } from '../../../helpers';
+
+export type NotificationListPreviewState = 'loading' | 'loaded with data' | 'loaded with no data';
 
 export interface NotificationListProps {
-	notificationType?: NotificationType,
-	previewState?: NotificationListPreviewState
-	innerRef?: React.Ref<HTMLDivElement>
+    previewState?: NotificationListPreviewState;
+    notificationType?: NotificationType;
+    notificationIdentifierPattern?: RegExp;
+    displayLimit?: number;
+    hideWhenEmpty?: boolean;
+    innerRef?: React.Ref<HTMLDivElement>;
 }
 
-export type NotificationListPreviewState = "Default" | "NoData";
+/**
+ * A list of participant notifications.
+ *
+ * <ul>
+ * <li>The list can be filtered by type (i.e. Email/Push/SMS) and/or identifier pattern.</li>
```

Comment:
Just double checking - does "pattern" imply regular expression in this type of usage?  I don't have a good sense as to how interchangable the phrase "pattern" and "regular expression" are.

Reply:
I think `pattern` is a fairly common terminology to use when referring to regular expressions.

Some examples here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/RegExp#return_value

However, I'm also fine with renaming to something like `notificationIdentifierRegex`, if you think that would be better understood.  Originally, I was going to have that property be a `string`, but then the component would have needed to handle invalid regex strings.  I figured it was better to leave that to the calling code so it could choose how to handle invalid regexes.  🙂